pkg/cloudevents/transport/http/transport_test.go

```diff
@@ -0,0 +1,128 @@
+package http_test
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/cloudevents/sdk-go/pkg/cloudevents"
+	cehttp "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/http"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/types"
+	"golang.org/x/sync/errgroup"
+)
+
+type task func() error
+
+// We can't use net/http/httptest.Server here because it's connection
+// tracking logic interferes with the connection lifecycle under test
+func startTestServer(handler http.Handler) (*http.Server, error) {
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return nil, err
+	}
+	server := &http.Server{
+		Addr:    listener.Addr().String(),
+		Handler: handler,
+	}
+	go server.Serve(listener)
+	return server, nil
+}
+
+func doConcurrently(concurrency int, duration time.Duration, fn task) error {
+	var group errgroup.Group
+	for i := 0; i < concurrency; i++ {
+		group.Go(func() error {
+			done := time.After(duration)
+			for {
+				select {
+				case <-done:
+					return nil
+				default:
+					if err := fn(); err != nil {
+						return err
+					}
+				}
+			}
+		})
+	}
+
+	if err := group.Wait(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// An example of how to make a stable client under sustained
+// concurrency sending to a single host
+func makeStableClient(addr string) (*cehttp.Transport, error) {
+	ceClient, err := cehttp.New(cehttp.WithTarget(addr))
+	if err != nil {
+		return nil, err
+	}
+	netHTTPTransport := &http.Transport{
+		MaxIdleConnsPerHost: 1000,
```

Comment:
@bbrowning, how does `MaxIdleConnsPerHost ` with a value of `1000` interact with `MaxIdleConns` with its default value of `100`?

My understanding is that `MaxIdleConns` is the number of connections to keep-alive across all hosts and `MaxIdleConnsPerHost` is the number of connections to keep alive per host. I'm guessing that if `MaxIdleConnsPerHost` has a higher value than `MaxIdleConns`, only `MaxIdleConns` will be used??

Reply:
@dubee I had to dig into the source to figure this out, but both values get applied. It first checks `MaxIdleConnsPerHost` and then enforces `MaxIdleConns` by evicting the oldest connections from the cache. There are slightly different semantics in the errors returned between the two limits as well as the actual behavior - hitting the global limit still puts the new connection in the cache and just evicts the oldest. Hitting the per-host limit doesn't put the new connection in the cache at all.

For the purposes of this test, that doesn't seem to actually matter, but it's good to know when tweaking these values on the Knative side of things.